packages/components/icons/src/leading-icon/leading-icon.styles.ts

```diff
@@ -0,0 +1,156 @@
+import { designTokens } from '@commercetools-uikit/design-system';
+import { css } from '@emotion/react';
+import { type TLeadingIconProps } from './leading-icon';
+
+const sizeMap = {
+  10: designTokens.spacing50,
+  20: `calc(${designTokens.spacing50} + ${designTokens.spacing20})`,
+  30: designTokens.spacing60,
+  40: designTokens.spacing70,
+};
+
+const colorThemeMap = {
+  accent: {
+    light: {
+      fill: designTokens.colorAccent40,
+      background: designTokens.colorAccent90,
+    },
+    dark: {
+      background: designTokens.colorAccent40,
+      fill: designTokens.colorNeutral90,
+    },
+  },
+  brown: {
+    light: {
+      background: designTokens.colorBrown90,
+      fill: designTokens.colorBrown50,
+    },
+    dark: {
+      background: designTokens.colorBrown50,
+      fill: designTokens.colorNeutral90,
+    },
+  },
+  customSvg: {
+    light: {
+      background: designTokens.colorSurface,
+      border: `solid ${designTokens.borderWidth1} ${designTokens.colorNeutral90}`,
+      fill: 'inherit',
+    },
+    dark: {
+      background: designTokens.colorSurface,
+      border: `solid ${designTokens.borderWidth1} ${designTokens.colorNeutral90}`,
+      fill: 'inherit',
+    },
+  },
+  neutral: {
+    light: {
+      background: designTokens.colorNeutral90,
+      fill: designTokens.colorNeutral40,
+    },
+    dark: {
+      background: designTokens.colorNeutral40,
+      fill: designTokens.colorNeutral90,
+    },
+  },
+  purple: {
+    light: {
+      background: designTokens.colorPurple90,
+      fill: designTokens.colorPurple50,
+    },
+    dark: {
+      background: designTokens.colorPurple50,
+      fill: designTokens.colorNeutral90,
+    },
+  },
+  turquoise: {
+    light: {
+      background: designTokens.colorTurquoise90,
+      fill: designTokens.colorTurquoise50,
+    },
+    dark: {
+      background: designTokens.colorTurquoise50,
+      fill: designTokens.colorNeutral90,
+    },
+  },
+  white: {
+    light: {
+      background: designTokens.colorSurface,
+      border: `solid ${designTokens.borderWidth1} ${designTokens.colorNeutral90}`,
+      fill: designTokens.colorNeutral60,
+    },
+    dark: {
+      background: designTokens.colorSolid,
+      fill: designTokens.colorNeutral90,
+    },
+  },
+};
+
+function getSizeStyles(size: TLeadingIconProps['size']): {
+  height: string;
+  width: string;
+} {
+  return {
+    height: sizeMap[size!],
+    width: sizeMap[size!],
+  };
+}
+
+function getColorThemeStyles({
+  color,
+  isInverted,
+}: {
+  color: TLeadingIconProps['color'] | 'customSvg';
+  isInverted: TLeadingIconProps['isInverted'];
+}): { background: string; fill?: string; border?: string; margin?: string } {
+  return colorThemeMap[color!][isInverted ? 'dark' : 'light'];
+}
```

Comment:
**nit**: these two functions are one liners, so perhaps we don't need them and can include those two lines in the `getLeadingIconStyles` function below.

Reply:
@CarlosCortizasCT these are declared as discrete functions so that the types are more readable and (in my opinion) easier to reason about.  I can change it up if you really prefer, but I'd just as soon leave it as-is.
![Screenshot 2024-01-30 at 8 02 11 AM](https://github.com/commercetools/ui-kit/assets/87667330/44479422-c82e-4f7e-9771-f3bc0b3e4f24)

Reply:
updated [here](https://github.com/commercetools/ui-kit/pull/2701/commits/ec6296c92de660bc69dbf14453079f7433827243)